.github/workflows/package.yml

```diff
@@ -1,27 +1,35 @@
 on:
-  push:
-    branches:
-      - master
+  pull_request:
+    types:
+      - opened
+      - edited
+      - synchronize
 
 name: Package
 
 jobs:
   check:
+    if: github.event.commits[0].author.name != 'GitHub Actions'
     name: Package distribution file
     runs-on: ubuntu-latest
     steps:
-    - name: Checkout
+    - name: Init a git repo
       uses: actions/checkout@v2
-      with:
-        ref: master
+    - name: Checkout PR
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: gh pr checkout ${{ github.event.pull_request.number }}
     - name: Package
       run: |
         npm ci
         npm test
         npm run package
     - name: Commit
+      if: github.event.pull_request.author.name == 'dependabot'
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       run: |
         git config --global user.name "GitHub Actions"
         git add dist/
         git commit -m "chore: Update dist" || echo "No changes to commit"
-        git push origin HEAD:master
+        git push
```

Comment:
hmmm.....i don't think we want to actually push the dist folder every time a PR is opened/changed, only after the PR is approved and merged to master? 🤔 i wonder if we should just remove the rule for this action? 🤔